lib/recipe.rb

```diff
@@ -0,0 +1,24 @@
+class Recipe 
+  attr_reader :name,
+              :ingredients_required
+  
+  def initialize(name)
+    @name                 = name
+    @ingredients_required = Hash.new(0)
+  end
+
+  def add_ingredient(ingredient, qty)
+    @ingredients_required[ingredient] += qty if @ingredients_required[ingredient] ||
+    @ingredients_required[ingredient] = qty
```

Comment:
This shouldn't be need since you have the default value. Just like the hash you have in the pantry class.